nixos/modules/services/matrix/conduwuit.nix

```diff
@@ -0,0 +1,245 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.conduwuit;
+
+  format = pkgs.formats.toml { };
+  # TOML does not allow null values, so we just omit those fields
+  filteredSettings = lib.converge (lib.filterAttrsRecursive (_: v: v != null)) cfg.settings;
+  configFile = format.generate "conduwuit.toml" filteredSettings;
+in
+{
+  meta.maintainers = with lib.maintainers; [ niklaskorz ];
+  options.services.conduwuit = {
+    enable = lib.mkEnableOption "conduwuit";
+
+    extraEnvironment = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      description = "Extra Environment variables to pass to the conduwuit server.";
+      default = { };
+      example = {
+        RUST_BACKTRACE = "yes";
+      };
+    };
+
+    package = lib.mkPackageOption pkgs "conduwuit" { };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = format.type;
+        options = {
+          global.server_name = lib.mkOption {
+            type = lib.types.str;
+            example = "example.com";
+            description = "The server_name is the name of this server. It is used as a suffix for user # and room ids.";
+          };
+          global.address = lib.mkOption {
+            type = lib.types.nullOr (
+              lib.types.oneOf [
+                lib.types.str
+                (lib.types.listOf lib.types.str)
+              ]
+            );
+            default = null;
+            description = ''
+              Address (IPv4 or IPv6) to listen on for connections by the reverse proxy/tls terminator.
+              If set to `null`, conduwuit will listen on IPv4 and IPv6 localhost.
+              Must be `null` if `unix_socket_path` is set.
+
+              To listen on multiple addresses, specify a list. For example:
+              `[ "127.0.0.1" "::1" ]`
+            '';
+          };
+          global.port = lib.mkOption {
+            type = lib.types.oneOf [
+              lib.types.port
+              (lib.types.listOf lib.types.port)
+            ];
+            default = 6167;
+            description = "The port(s) conduwuit will be running on. You need to set up a reverse proxy in your web server (e.g. apache or nginx), so all requests to /_matrix on port 443 and 8448 will be forwarded to the conduwuit instance running on this port";
+          };
+          global.unix_socket_path = lib.mkOption {
+            type = lib.types.nullOr lib.types.str;
+            default = null;
+            description = ''
+              Listen on a UNIX socket at the specified path. If listening on a UNIX socket,
+              listening on an address will be disabled. The `address` option must be set to
+              `null` (the default value). You must add your reverse proxy to the `conduwuit`
+              group, unless RW permissions are specified with `unix_socket_perms` (666 minimum).
```

Comment:
```suggestion
              group.
              
              This will automatically disable the `DynamicUser` systemd feature and add a static user/group
              to your system since this is the intended way how unix sockets should be used with conduwuit.
```